core/http.md

```diff
@@ -0,0 +1,3293 @@
+
+When `$details` is used, or the Resource is not configured to have a
+domain-specific document, then a successful response MUST be of the form:
+
+A successful response MUST be of the form:
+
+```yaml
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+
+{
+  ... Version entity excluded for brevity ...
+}
+```
+
+When `$details` is not used and the Resource is configured to have a
+domain-specific document, then a successful response MUST either return the
+Version's domain-specific document, or an HTTP redirect to
+the `<RESOURCE>url` value if set.
+
+In the case of returning the Version's domain-specific document, the response
+MUST be of the form:
+
+```yaml
+HTTP/1.1 200 OK
+or
+HTTP/1.2 303 See Other
+Content-Type: <STRING> ?
+xRegistry-<RESOURCE>id: <STRING>
+xRegistry-versionid: <STRING>
+xRegistry-self: <URL>
+xRegistry-xid: <XID>
+xRegistry-epoch: <UINTEGER>
+xRegistry-name: <STRING> ?
+xRegistry-isdefault: <BOOLEAN>
+xRegistry-description: <STRING> ?
+xRegistry-documentation: <URL> ?
+xRegistry-icon: <URL> ?
+xRegistry-labels-<KEY>: <STRING> *
+xRegistry-createdat: <TIMESTAMP>
+xRegistry-modifiedat: <TIMESTAMP>
+xRegistry-ancestor: <STRING>
+Location: <URL> ?                        # If 303 is returned
+Content-Disposition: <STRING> ?
+
+... Version document ...                 # If <RESOURCE>url is not set
+```
+
+Where:
+- `Content-Disposition` SHOULD be present and if so, MUST be the `<RESOURCE>id`
+  value. This allows for HTTP tooling that is not aware of xRegistry to know
+  the desired filename to use if the HTTP body were to be written to a file.
+- `Location`, if present, and `<RESOURCE>url` MUST have the same value.
+
+**Examples:**
+
+Retrieve a specific Version of a `schema` Resource as xRegistry metadata:
+
+```yaml
+GET /schemagroups/g1/schemas/myschema/versions/1.0$details
+```
+
+```yaml
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+
+{
+  "schemaid": "myschema",
+  "versionid": "1.0",
+  "self": "https://example.com/schemagroups/g1/schemas/myschema/versions/1.0",
+  "xid": "/endpoints/ep1/messages/msg1/versions/1.0",
+  "epoch": 2,
+  "isdefault": true,
+  "createdat": "2024-04-30T12:00:00Z",
+  "modifiedat": "2024-04-30T12:00:01Z",
+  "ancestor": "1.0"
+}
+```
+
+Retrieve the domain-specific document of the same schema Resource:
+
+```yaml
+GET /schemagroups/g1/schemas/myschema/versions/1.0
+```
+
+```yaml
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+xRegistry-schemaid: myschema
+xRegistry-versionid: 1.0
+xRegistry-self: https://example.com/schemagroups/g1/schemas/myschema/versions/1.0
+xRegistry-xid: /schemagroups/g1/schemas/myschema/versions/1.0
+xRegistry-epoch: 2
+xRegistry-isdefault: true
+xRegistry-createdat: 2024-04-30T12:00:00Z
+xRegistry-modifiedat: 2024-04-30T12:00:01Z
+xRegistry-ancestor: 1.0
+Content-Disposition: myschema
+
+{ ... Contents of a schema doc excluded for brevity ...  }
+```
+
+#### `PATCH /<GROUPS>/<GID>/<RESOURCES>/<RID>/versions/<VID>`]
```

Comment:
Not sure what these `]` end-of-line characters are doing here

Reply:
I think those used to have some flags listed at the end. Done